test/js/explore/plot-data-cache.test.js

```diff
@@ -0,0 +1,329 @@
+import * as ScpApi from 'lib/scp-api'
+
+import _cloneDeep from 'lodash/cloneDeep'
+import { createCache } from 'components/explore/plot-data-cache'
+
+// models a real response from api/v1/visualization/clusters
+const FETCH_CLUSTER_RESPONSE = {
+    data: {
+        annotations: ['foo', 'bar'],
+        cells: ['A', 'B'],
+        x: [11, 14],
+        y: [0, 1],
+    },
+    pointSize: 3,
+    userSpecifiedRanges: null,
+    showClusterPointBorders: false,
+    description: null,
+    is3D: false,
+    isSubsampled: false,
+    isAnnotatedScatter: false,
+    numPoints: 130,
+    axes: {
+        titles: {
+            x: 'X',
+            y: 'Y',
+            z: 'Z',
+            magnitude: 'Expression'
+        },
+        aspects: null
+    },
+    hasCoordinateLabels: false,
+    coordinateLabels: [],
+    pointAlpha: 1,
+    cluster: 'cluster.tsv',
+    genes: [],
+    annotParams: {
+        name: 'buzzwords',
+        type: 'group',
+        scope: 'study',
+        values: ['foo', 'bar'],
+        identifier: 'biosample_id--group--study'
+    },
+    subsample: 'all',
+    consensus: null
+}
+
+const CACHE_PERF_PARAMS = {
+  legacyBackend: 0,
+  parse: 0,
+  url: 'cache'
+}
+
+
+const ANNOTATION_ONLY_RESPONSE = _cloneDeep(FETCH_CLUSTER_RESPONSE)
+ANNOTATION_ONLY_RESPONSE.annotParams = {
+    name: 'species',
+    type: 'group',
+    scope: 'study',
+    values: ['dog', 'cat'],
+    identifier: 'species--group--study'
+  }
+ANNOTATION_ONLY_RESPONSE.data = {
+  'annotations': ['cat', 'dog']
+}
+
+const EXPRESSION_ONLY_RESPONSE = _cloneDeep(ANNOTATION_ONLY_RESPONSE)
+EXPRESSION_ONLY_RESPONSE.data = {
+  'expression': [0.25, 2.3],
+}
+EXPRESSION_ONLY_RESPONSE.genes = ['Apoe']
+
+const CLUSTER_AND_EXPRESSION_RESPONSE = _cloneDeep(FETCH_CLUSTER_RESPONSE)
+CLUSTER_AND_EXPRESSION_RESPONSE.data.expression = [0.25, 2.3]
+CLUSTER_AND_EXPRESSION_RESPONSE.genes = ['Apoe']
+
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
+describe('Plot data cache', () => {
+  it('caches a single cluster call for use when annotation and gene are changed', async () => {
+    const cache = createCache()
+    const apiFetch = jest.spyOn(ScpApi, 'fetchCluster')
+    // pass in a clone of the response since it may get modified by the cache operations
+    apiFetch.mockImplementation(() => Promise.resolve([_cloneDeep(FETCH_CLUSTER_RESPONSE), 230]))
```

Comment:
What does the `230` map to?  Is that the `perfTime` calculation?

Reply:
yeah, so for purposes of this test it just needs to be any number/object